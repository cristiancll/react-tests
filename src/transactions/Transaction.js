import React from 'react';
import PropTypes from 'prop-types';

import './Transaction.css'


function Transaction({ date, type, value }) {
    return (
        <div className="transaction-container">
            <p>{date}</p>
            <p>{type}</p>
            <p>R$ {value}</p>
        </div>
    )
};

Transaction.defaultProps = {
    type: '',
    value: '0',
    date: '',
}

Transaction.propTypes = {
    type: PropTypes.string,
    value: PropTypes.string,
    date: PropTypes.string,
}

export default Transaction;