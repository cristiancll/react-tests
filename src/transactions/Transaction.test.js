import React from "react";
import Transaction from "./Transaction";
import {render, screen} from "@testing-library/react";

describe('Componente de transação do extrato', () => {
    it('O snapshot do componente deve permanecer sempre o mesmo', () => {
        const {container} = render(<Transaction date={"08/09/2020"} type={"withdrawn"} value={"20.00"}/>);
        expect(container.firstChild).toMatchSnapshot();
    });
}); 