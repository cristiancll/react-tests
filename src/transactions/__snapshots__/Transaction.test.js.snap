// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de transação do extrato O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  class="transaction-container"
>
  <p>
    08/09/2020
  </p>
  <p>
    withdrawn
  </p>
  <p>
    R$ 
    20.00
  </p>
</div>
`;
